File: 2553/2553.c
int separateTheDigitsInAnArray(struct TreeNode* root) {
    if (!root) return 0;
    int l = separateTheDigitsInAnArray(root->left);
    int r = separateTheDigitsInAnArray(root->right);
    return 1 + (l > r ? l : r);
}

File: 2553/2553.py
class Solution:
    def separate_the_digits_in_an_array(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.separate_the_digits_in_an_array(root.left), self.separate_the_digits_in_an_array(root.right))